feat..Detail page was added with author information and comments section

BayramovaKonul opened the pull request:
Url configuartions have been set up for each article's details, author info and comment section. Blog, categories and details pages are fully functional.

The search in the category page doesn't work as intended. Text in the detail page is unreadable.

BayramovaKonul committed:
feat..Detail page was added with author information and comments section

Url configuartions have been set up for each article's details, author info and comment section. Blog, categories and details pages are fully functional

valiyevvali commented on blog/urls.py:
```diff
@@ -8,5 +8,5 @@
     path('blogs', views.blogs, name = 'blogs'),
     path('category/<slug:category_slug>', views.category_blog, name = 'category'),
-    path('blogs/<int:id>/', views.detail, name = 'blogs/<int:id>/'),
+    path('blogs/<int:blog_id>/', views.detail, name = 'blog/details'),
```
dynamic part must be blog's slug not id. For SEO oriented urls

valiyevvali commented on blog/views.py:
```diff
@@ -40,12 +40,34 @@ def blogs (request):
-    return render(request, 'detail.html')
\ No newline at end of file
+
+def detail (request,blog_id):
```
too general function name

valiyevvali commented on blog/views.py:
```diff
@@ -40,12 +40,34 @@ def blogs (request):
\ No newline at end of file
+
+def detail (request,blog_id):
+    details = ArticleModel.objects.get(id = blog_id)
```
Check maybe blog with this id(replace it with slug) does not exist

BayramovaKonul replied:
we go to article_detail using its display on the page (such as articles on blog page). If it doesn't exist, how can it be shown on the page?

valiyevvali replied:
You are right, but we can manually write `articles/random_slug` in browser tab ,in that case your code will not work properly.
But If you handle this case , you can show 404 page to user which is correct option.

valiyevvali commented on blog/views.py:
```diff
@@ -40,12 +40,34 @@ def blogs (request):
+    blogs = ArticleModel.objects.all()
+    recent_posts = blogs.order_by('-created_at')[:4]
+    categories = CategoryModel.objects.annotate(article_count=Count('articles')).order_by("-article_count").values('name', 'article_count', 'slug')
+    return render(request, 'detail.html', context={
```
template name (detail.html) is too common.Can be `blog_detail`

valiyevvali commented on blog/views.py:
```diff
@@ -40,12 +40,34 @@ def blogs (request):
-    categories = get_object_or_404(CategoryModel, slug = category_slug)  # check the table to find asked category_slug
-    category_blogs = categories.articles.all() # get all articles from table based on the category using related name
+    categories = CategoryModel.objects.annotate(article_count=Count('articles')).order_by("-article_count").values('name', 'article_count', 'slug')
+    slug_categories = get_object_or_404(CategoryModel, slug = category_slug)  # check the table to find asked category_slug
```
naming is not correct, it is returning 1 category object so it should be singular. can be `category_obj`

valiyevvali commented:
Remove __py_cache__ folders and add it to gitignore

BayramovaKonul committed:
details_comments_author

Blog_detail page  has been created using slug, names were editted, except pycache

valiyevvali committed:
Merge branch 'main' into feature/details_comments_author
